Sources/MetadataHDWalletKit/Wallet/PrivateKey.swift

```diff
@@ -232,20 +233,41 @@ extension PrivateKey {
 
         func serialize() -> Data {
             var bytes = [UInt8]()
+            // version: puts xprv or xpub at the start
             bytes += Data(from: coin.privateKeyVersion.bigEndian).bytes
+            // depth: how many times this child has been derived from master key (0 = master key)
             bytes += [depth]
+            // fingerprint: created from parent public key (allows you to spot adjacent xprv and xpubs)
             bytes += Data(from: parentFingerprint).bytes
+            // index: the index of this child key from the parent
             bytes += index.bytes
+            // chainCode: the current chain code being used for this key
             bytes += chainCode
+            // the private key you want to create a serialized extended key for (prepend 0x00 for private)
             bytes += getPrivKeyBytes33()
             assert(bytes.count == 78)
             return Data(bytes)
         }
 
-        func addChecksum(data: Data) -> Data {
-            var data = data
-            data += data.doubleSHA256.prefix(4)
-            return data
+        let serialized = serialize()
+        let checksummed = addChecksum(data: serialized)
+        let encoded = Base58.encode(checksummed)
+        return encoded
+    }
+
+    /// Provides the extended public key
+    public func extendedPublic() -> String {
+
+        func serialize() -> Data {
+            var bytes = [UInt8]()
```

Comment:
Could we add here the same comments that we have `func serialize() -> Data`? Will make it easier to read.

Reply:
yeah i could add the "same" comments